packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel_cloud_firestore.dart

```diff
@@ -0,0 +1,356 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show visibleForTesting, required;
+
+import '../cloud_firestore_platform_interface.dart';
+
+class MethodChannelCloudFirestore extends CloudFirestorePlatform {
+  MethodChannelCloudFirestore() {
+    channel.setMethodCallHandler(_handlePlatformCall);
+  }
+
+  @visibleForTesting
+  static const MethodChannel channel = MethodChannel(
+    'plugins.flutter.io/cloud_firestore',
+  );
+
+  // Platform calls
+  Future<void> _handlePlatformCall(MethodCall call) async {
+    switch(call.method) {
+      case 'QuerySnapshot':
+        return _handleQuerySnapshot(call);
+        break;
+      // case 'DocumentSnapshot':
+      //   final DocumentSnapshot snapshot = DocumentSnapshot._(
+      //     call.arguments['path'],
+      //     _asStringKeyedMap(call.arguments['data']),
+      //     SnapshotMetadata._(call.arguments['metadata']['hasPendingWrites'],
+      //         call.arguments['metadata']['isFromCache']),
+      //     this,
+      //   );
+      //   _documentObservers[call.arguments['handle']].add(snapshot);
+      //   break;
+      case 'DoTransaction':
+        return _handleDoTransaction(call);
+        break;
+    }
+  }
+
+  // Global
+  @override
+  Future<void> removeListener(int handle) =>
+     channel.invokeMethod<void>(
+      'removeListener',
+      <String, dynamic>{'handle': handle},
+    );
+
+
+  // Firestore
+  @override
+  Future<void> enablePersistence(String app, {bool enable = true}) => 
+    channel
+        .invokeMethod<void>('Firestore#enablePersistence', <String, dynamic>{
+      'app': app,
+      'enable': enable,
+    });
+  
+
+  @override
+  Future<void> settings(String app, {
+      bool persistenceEnabled,
+      String host,
+      bool sslEnabled,
+      int cacheSizeBytes,
+    }) =>
+     channel.invokeMethod<void>('Firestore#settings', <String, dynamic>{
+      'app': app,
+      'persistenceEnabled': persistenceEnabled,
+      'host': host,
+      'sslEnabled': sslEnabled,
+      'cacheSizeBytes': cacheSizeBytes,
+    });
+
+  // Transaction data
+  static final Map<int, PlatformTransactionHandler> _transactionHandlers =
+      <int, PlatformTransactionHandler>{};
+  static int _transactionHandlerId = 0;
+
+  @override
+  Future<Map<String, dynamic>> runTransaction(String app, {
+    @required PlatformTransactionHandler transactionHandler,
+    int transactionTimeout,
+  }) async {
+    // Store the transaction handler function so we can use it later on DoTransaction...
```

Comment:
This comment seems unnecessary to me

Reply:
Yeah, I'll rewrite, but keep a mention to the `_handleDoTransaction` method. 

It's not easy to see that `runTransaction` is split between here and the result from `_handleDoTransaction` :)